### DISCUSSION TITLE
Installing a patch that includes changes to non existing files

### DISCUSSION THREAD

**Discussion by robdekort** [cweagans/composer-patches#528] [Q&A] [answered]
First of all: thanks for this great package. I use it a lot to pull in unreleased PR's for Statamic (a CMS). However some PR's prove difficult to pull in as they include PHP tests that aren't included in the actual release build that gets installed and is the patch target. 

Those patches will fail to install and the solution is to manually edit the patch file to remove all references to tests. This can be a very tedious process. I'm wondering if there's a way for this package to skip and ignore files that don't exist?

Thanks a bunch in advance.

**Comment by cweagans** [comment, marked as answer]
You could potentially use the [freeform patcher](https://docs.cweagans.net/composer-patches/usage/freeform-patcher/) functionality in 2.x. I'd also be open to a PR that adds a way to pass args to specific patchers. In this case, you'd probably want the `--exclude` flag for `git apply`.

Another possibility is to semi-automate this along these lines:

```bash
git clone [your repo]
cd [your repo]
git apply --exclude [test paths] [your patch]
git diff > patch-without-test-paths.patch
```

And then use `patch-without-test-paths.patch` in your project instead. This is still a manual process, but it could be automated with a script. If you do that, you could potentially run that script on a Composer hook.

**Reply by robdekort** [reply]
Hey Cameron. Thanks for your answer. I'm gonna try your suggestions next time. I'm afraid PR'ing your package is not something within my skillset.
